### PR TITLE
[FrameworkBundle] Configure `logger` as error logger if the Monolog Bundle is not registered

### LINKED ISSUE #51910: [ErrorHandler] Loggers can only be Monolog’s
### Symfony version(s) affected

≥ 5.1

### Description

#36621 replaced `DebugHandlersListener`’s loggers for Monolog’s ones. Then, #49275 introduced the `ErrorHandlerConfigurator` but its loggers are configured the same way.

As a result, HttpKernel’s `Logger` won’t be passed to the `ErrorHandler`, even if you’re not using Monolog. That means eg. deprecations won’t appear in the profiler’s logs panel, but they will always appear in commands output (because `ErrorHandler`’s loggers will be `BufferingLogger`s).

### How to reproduce

In a Symfony project without `symfony/monolog-bundle` using the profiler, trigger an error or deprecation and it won’t appear in the profiler’s logs panel.

### Possible Solution

`ErrorHandlerConfigurator::$logger` and `ErrorHandlerConfigurator::$deprecationLogger` could be injected the `logger` service by default and only be wired to `monolog.logger.php` or `monolog.logger.deprecation` if they exist.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #51910
| License       | MIT

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ErrorLoggerCompilerPass.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+class ErrorLoggerCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('debug.debug_handlers_listener')) {
+            return;
+        }
+
+        $definition = $container->getDefinition('debug.error_handler_configurator');
```
Comment (HypeMC) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ErrorLoggerCompilerPass.php]: The `debug.error_handler_configurator` service was added in 6.3, this targets 5.4.

Comment (MatTheCat) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ErrorLoggerCompilerPass.php]: Indeed forgot to update that one, thanks

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ErrorLoggerCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class ErrorLoggerCompilerPass implements CompilerPassInterface
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ErrorLoggerCompilerPass.php:21]: let's mark it `@internal`

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php [hunk 3]
```diff
@@ -536,7 +536,7 @@ public function testPhpErrorsWithLogLevels()
         $container = $this->createContainerFromFile('php_errors_log_levels');
 
         $definition = $container->getDefinition('debug.debug_handlers_listener');
-        $this->assertEquals(new Reference('monolog.logger.php', ContainerInterface::NULL_ON_INVALID_REFERENCE), $definition->getArgument(1));
+        $this->assertEquals(new Reference('logger', ContainerInterface::NULL_ON_INVALID_REFERENCE), $definition->getArgument(1));
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php:539]: do we still need `NULL_ON_INVALID_REFERENCE` on all 3 modified lines?

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php [hunk 4]
```diff
@@ -21,12 +21,12 @@
...
             ->args([
                 null, // Exception handler
-                service('monolog.logger.php')->nullOnInvalid(),
+                service('logger')->nullOnInvalid(),
                 null, // Log levels map for enabled error levels
                 param('debug.error_handler.throw_at'),
                 param('kernel.debug'),
                 param('kernel.debug'),
-                service('monolog.logger.deprecation')->nullOnInvalid(),
+                service('logger')->nullOnInvalid(),
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php:29]: do we still need nullOnInvalid on modified lines?

Comment (MatTheCat) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php:29]: HttpKernel’s `LoggerPass` should make sure the `logger` service exist, but I guess it is possible to override this behavior in userland?

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php (hunk 3 above)
Comment (MatTheCat) [reply src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php:539]: I guess this depends on how https://github.com/symfony/symfony/pull/52009#discussion_r1356803917 turns out :eyes:

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php (hunk 4 above)
Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php:29]: alright, let's keep as is

**Comment by nicolas-grekas** [comment]
Thank you @MatTheCat.

**Comment by PhilETaylor** [comment]
hmmm - got this after composer update - remains after manually deleting caches. 
ScreenShot-2023-10-12-17 40 51

ScreenShot-2023-10-12-17 38 24

**Comment by PhilETaylor** [comment]
I should be clear, on 6.4.* 

```
 - Downloading symfony/http-kernel (6.4.x-dev 33464ce)
  - Downloading symfony/validator (6.4.x-dev dbd0115)
  - Downloading symfony/http-foundation (6.4.x-dev 38dc0de)
  - Downloading symfony/security-http (6.4.x-dev bcf2cc2)
  - Downloading symfony/security-bundle (6.4.x-dev 8010762)
  - Downloading symfony/framework-bundle (6.4.x-dev b7abd18)
  - Downloading symfony/lock (6.4.x-dev c743cb7)
  - Downloading symfony/cache (6.4.x-dev f550e6d)
  - Downloading symfony/doctrine-bridge (6.4.x-dev f804b48)
  - Downloading friendsofphp/php-cs-fixer (v3.35.1)
  - Downloading phpstan/phpstan (1.11.x-dev 22cb9ce)
  - Upgrading symfony/http-kernel (6.4.x-dev a4425b6 => 6.4.x-dev 33464ce): Extracting archive
  - Upgrading symfony/validator (6.4.x-dev a5956e9 => 6.4.x-dev dbd0115): Extracting archive
  - Upgrading symfony/http-foundation (6.4.x-dev 688f1a9 => 6.4.x-dev 38dc0de): Extracting archive
  - Upgrading symfony/security-http (6.4.x-dev fc7a711 => 6.4.x-dev bcf2cc2): Extracting archive
  - Upgrading symfony/security-bundle (6.4.x-dev 8ba82a2 => 6.4.x-dev 8010762): Extracting archive
  - Upgrading symfony/framework-bundle (6.4.x-dev f9ee22f => 6.4.x-dev b7abd18): Extracting archive
  - Upgrading symfony/lock (6.4.x-dev b0878f1 => 6.4.x-dev c743cb7): Extracting archive
  - Upgrading symfony/cache (6.4.x-dev cf85501 => 6.4.x-dev f550e6d): Extracting archive
  - Upgrading symfony/doctrine-bridge (6.4.x-dev 1618825 => 6.4.x-dev f804b48): Extracting archive
  - Upgrading friendsofphp/php-cs-fixer (v3.35.0 => v3.35.1): Extracting archive
  - Upgrading phpstan/phpstan (1.11.x-dev 348b72e => 1.11.x-dev 22cb9ce): Extracting archive
  - ```

**Comment by MatTheCat** [comment]
Seems like a rebase issue: `debug.error_handler_configurator` should be configured since 6.3. Ping @nicolas-grekas

**Comment by PhilETaylor** [comment]
I can see in vendor/symfony/framework-bundle/Resources/config/debug_prod.php

ScreenShot-2023-10-12-17 52 45

and if I add `dd(__FILE__)` in that function, the function is being invoked. 

Sorry above my internals knowledge here :)

**Comment by nicolas-grekas** [comment]
PR welcome to fix this. Merging has been hard today.

**Comment by PhilETaylor** [comment]
> PR welcome to fix this. Merging has been hard today.

I would not know what the fix is. Sorry to hear you have had a hard day with merging, Im totally not looking for support, or a speedy fix at all, just reporting what I found (I appreciate my dev-code is bleeding edge, and thus breaks often).  💰incoming :)

**Comment by nicolas-grekas** [comment]
Thanks for your support!

**Comment by PhilETaylor** [comment]
Maybe this? https://github.com/symfony/symfony/pull/52023

**Pull request merged**